src/client/components/app/Transactions/Labs/LabDepositTx.tsx

```diff
@@ -110,17 +121,14 @@ export const LabDepositTx: FC<LabDepositTxProps> = ({ onClose }) => {
     return null;
   }
 
-  let isApproved: boolean | undefined;
-  let allowanceError: string | undefined;
-
-  const { approved, error } = basicValidateAllowance({
-    tokenAddress: selectedSellTokenAddress,
-    tokenAmount: toBN(debouncedAmount),
-    tokenDecimals: selectedSellToken.decimals.toString(),
-    rawAllowance: tokenAllowance?.amount || '0',
+  const { approved: isApproved, error: allowanceError } = validateVaultAllowance({
```

Comment:
How about here just changing the return from `validateVaultAllowance()` as it appears we were not and still aren't happy with the naming?

Reply:
not following the issue here. can you explain? not sure if we should rename the validation function name

Reply:
Sorry I wasn't very clear, what I meant was to change the returned object keys, so instead of `approve` we would return `isApproved` from the `validateVaultAllowance()` as it seems that's the name we want and it's best practice when returning a boolean to prefix with "is"

Reply:
Let's leave as is for now so we can resolve the bug and maybe then get back to this later. Resolving